bundles/org.eclipse.orion.client.debug/web/orion/debug/nativeDeployService.js

```diff
@@ -0,0 +1,242 @@
+/*******************************************************************************
+ * @license
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0
+ * (http://www.eclipse.org/legal/epl-v10.html), and the Eclipse Distribution
+ * License v1.0 (http://www.eclipse.org/org/documents/edl-v10.html).
+ *
+ * Contributors: IBM Corporation - initial API and implementation
+ ******************************************************************************/
+/*eslint-env browser, amd*/
+define([
+    'i18n!orion/debug/nls/messages',
+	'orion/Deferred',
+    'orion/debug/debugSocket',
+    'orion/debug/debugDumbHover'
+], function(messages, Deferred, mDebugSocket, mDebugDumbHover) {
+
+    'use strict';
+
+    /**
+     * A debug service provides breakpoints, watches and highlighted lines management.
+     * Here management means loading and saving data from storage, apis of getters and setters, and correspoding events.
+     * Note that breakpoints from streaming (remote-only) files won't be presisted.
+     * 
+     * @class {orion.debug.NativeDeployService}
+     * @param {orion.serviceregistry.ServiceRegistry} serviceRegistry
+     * @param {orion.commandregistry.CommandRegistry} commandRegistry
+     */
+    var NativeDeployService = function(serviceRegistry, commandRegistry) {
```

Comment:
I call the new deploy service for the debugger "Native deploy service". I don't think it's a good name, but other names I can imagine are even worse. Here are the reasons that I didn't select the following names:
Debug deploy service: You can also deploy a CF app with node-inspector and debug it.
Local deploy service: It's not local. The debugee runs on the same machine as Orion server runs.
Remote deploy service: All deployments are remote, aren't they? Also it confuse the "remote debug" checkbox in the configuration.